src/CommunityToolkit.Maui.SourceGenerators/Extensions/CompilationExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Extensions
+{
+	static class CompilationExtensions
+	{
+		public static TSymbol? GetSymbol<TSymbol>(this Compilation compilation, BaseTypeDeclarationSyntax declarationSyntax)
+			where TSymbol : ISymbol
+		{
+			var model = compilation.GetSemanticModel(declarationSyntax.SyntaxTree);
+			return (TSymbol?)model.GetDeclaredSymbol(declarationSyntax);
+		}
+	}
+}
```

Comment:
The added `﻿using Microsoft.CodeAnalysis.CSharp` will bring in an extension method from `Microsoft.CodeAnalysis.CSharp.CSharpExtensions`, which returns an `INamedTypeSymbol?`

This eliminates the need for a cast.

```suggestion
﻿using Microsoft.CodeAnalysis;
﻿using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;

namespace CommunityToolkit.Maui.SourceGenerators.Extensions
{
	static class CompilationExtensions
	{
		public static INamedTypeSymbol? GetSymbol(this Compilation compilation, BaseTypeDeclarationSyntax declarationSyntax)
		{
			var model = compilation.GetSemanticModel(declarationSyntax.SyntaxTree);
			return model.GetDeclaredSymbol(declarationSyntax);
		}
	}
}
```

Reply:
@Youssef1313 I think that you missed something here... This method is an extension method that can be used for other Symbols, like `ITypeSymbol`, `IFieldSymbol`, `IEventSymbol`, etc... So removing the Generic isn't a solution

Reply:
@pictos I thought there is a single caller only?

Anyway, there are multiple overloads of the `GetDeclaredSymbol` extension. So there is probably no need for the generic extension and all callers should call `model.GetDeclaredSymbol` directly (they just shouldn't miss the `using Microsoft.CodeAnalysis.CSharp;`)